azurecosmos/src/main/java/site/ycsb/db/AzureCosmosClient.java

```diff
@@ -40,120 +62,156 @@
+        "azurecosmos.maxRetryAttemptsOnThrottledRequests", retryOptions.getMaxRetryAttemptsOnThrottledRequests()));
+    retryOptions.setMaxRetryWaitTime(Duration.ofSeconds(this.getIntProperty("azurecosmos.maxRetryWaitTimeInSeconds",
+        Math.toIntExact(retryOptions.getMaxRetryWaitTime().toMillis() / 1000))));
+
+    DirectConnectionConfig directConnectionConfig = new DirectConnectionConfig();
+    directConnectionConfig.setMaxConnectionsPerEndpoint(this.getIntProperty("azurecosmos.maxConnectionPoolSize",
+        directConnectionConfig.getMaxConnectionsPerEndpoint()));
+    directConnectionConfig
+        .setIdleConnectionTimeout(Duration.ofSeconds(this.getIntProperty("azurecosmos.idleConnectionTimeout",
+            Math.toIntExact(directConnectionConfig.getIdleConnectionTimeout().toMillis() / 1000))));
+    directConnectionConfig.setMaxRequestsPerConnection(10000);
+
+    GatewayConnectionConfig gatewayConnectionConfig = new GatewayConnectionConfig();
+    gatewayConnectionConfig.setMaxConnectionPoolSize(
+        this.getIntProperty("azurecosmos.maxConnectionPoolSize", gatewayConnectionConfig.getMaxConnectionPoolSize()));
+    gatewayConnectionConfig
+        .setIdleConnectionTimeout(Duration.ofSeconds(this.getIntProperty("azurecosmos.idleConnectionTimeout",
+            Math.toIntExact(gatewayConnectionConfig.getIdleConnectionTimeout().toMillis() / 1000))));
 
     try {
-      LOGGER.info("Creating azurecosmos client {}.. connectivityMode={}, consistencyLevel={},"
-                      + " maxRetryAttemptsOnThrottledRequests={}, maxRetryWaitTimeInSeconds={}"
-                      + " useUpsert={}",
-              uri,
-              connectionPolicy.getConnectionMode(),
-              consistencyLevel.toString(),
-              connectionPolicy.getRetryOptions().getMaxRetryAttemptsOnThrottledRequests(),
-              connectionPolicy.getRetryOptions().getMaxRetryWaitTimeInSeconds(),
-              this.useUpsert);
-      AzureCosmosClient.client = new DocumentClient(uri, primaryKey, connectionPolicy, consistencyLevel);
-      LOGGER.info("Azure Cosmos connection created: {}", uri);
+      LOGGER.info(
+          "Creating Cosmos DB client {}, useGateway={}, consistencyLevel={},"
+              + " maxRetryAttemptsOnThrottledRequests={}, maxRetryWaitTimeInSeconds={}"
+              + " useUpsert={}, maxDegreeOfParallelism={}, maxBufferedItemCount={}, preferredPageSize={}",
+          uri, useGateway, consistencyLevel.toString(), retryOptions.getMaxRetryAttemptsOnThrottledRequests(),
+          retryOptions.getMaxRetryWaitTime().toMillis() / 1000, AzureCosmosClient.useUpsert,
+          AzureCosmosClient.maxDegreeOfParallelism, AzureCosmosClient.maxBufferedItemCount,
+          AzureCosmosClient.preferredPageSize);
+
+      CosmosClientBuilder builder = new CosmosClientBuilder().endpoint(uri).key(primaryKey)
+          .throttlingRetryOptions(retryOptions).endpointDiscoveryEnabled(false).consistencyLevel(consistencyLevel)
```

Comment:
any reason we are setting `endpointDiscoveryEnabled()` to false?

Reply:
V2 also had it as false, so I didn't modify it.